lib/router.ex

```diff
@@ -0,0 +1,22 @@
+defmodule HTTPServer.Router do
+  alias HTTPServer.Response
+  alias HTTPServer.Request
+
+  def route(%Request{method: "POST", path: "/echo_body"} = req) do
+    Response.build_response(req, 200)
+  end
+
+  def route(%Request{method: "GET", path: "/simple_get"} = req) do
+    Response.build_response(req, 200)
+  end
+
+  def route(%Request{method: "GET", path: "/simple_get_with_body"} = req) do
+    body = "Hello world"
+    req_with_body = %{req | body: body, headers: %{req.headers | "Content-Length" => "#{String.length(body)}"}}
+    Response.build_response(req_with_body, 200)
+  end
```

Comment:
This "works" but is it "right"?  This router doesn't seem to route at all, it appears to build responses.

Reply:
I see what you're saying, and I think I was questioning the implementation myself. I'm changing this now so that the only thing passed to build_response is the status code, and potentially any body that should be included as data in the response (almost like how you use res.send() to send your response in express.js).

Reply:
If you are going to introduce the concept of a router, is now also the time to introduce the concepts of handler and/or controllers? Does it make sense to route without them?

Reply:
I was looking into that a little bit yesterday evening, I think it felt unclear because there isn't a "source" of data that I would query in this instance. But I'm realizing that perhaps it could exist in the current state to just return a string and send it to the response? Or should I maybe create a file that's just a JSON object with some of the necessary "data" aka return strings that I can read from?

Reply:
This has been updated btw! Let me know if I'm close to getting it right. I think there might still be some flaws in the logic of how I've set it up, but I do think I've created greater separation between the router's responsibility and what the acceptance tests need to pass.